sr_gui_dynamic_plot_tool/src/sr_gui_dynamic_plot_tool/dynamic_plot_tool.py

```diff
@@ -0,0 +1,467 @@
+                selection_button_joint.setCheckable(True)
+                selection_button_joint.setText(str(joint[3:]))
+                selection_button_joint.setObjectName(str(joint[3:]))
+                selection_button_joint.released.connect(self._joint_button_released)
+                self.selection_button_joint.append(selection_button_joint)
+                sublayout_joint.addWidget(selection_button_joint)
+            subframe_joint.setLayout(sublayout_joint)
+            self.plot_interface_layout.addWidget(subframe_joint)
+
+    def _create_configuration_widget(self, name, parameters):
+        """
+        Create widgets for the entries in the interface settings
+        @param parameters - dict with type and entries for the widgets
+        """
+        sublayout_configuration = QtWidgets.QHBoxLayout()
+        sublayout_configuration.setAlignment(Qt.AlignLeft)
+        sublayout_configuration.setContentsMargins(0, 0, 0, 0)
+        subframe_configuration = QtWidgets.QFrame()
+        label_name = QtWidgets.QLabel()
+        label_name.setText("Select "+name)
+        label_name.setFont(QFont("Helvetica", 12))
+        self.plot_interface_layout.addWidget(label_name)
+        for field in parameters:
+            selection_button_configuration = QtWidgets.QToolButton()
+            selection_button_configuration.setFixedSize(150, 40)
+            selection_button_configuration.setCheckable(True)
+            selection_button_configuration.setText(field)
+            selection_button_configuration.setObjectName(field)
+            selection_button_configuration.released.connect(self._configuration_button_released)
+            self.selection_button_configuration.append(selection_button_configuration)
+            sublayout_configuration.addWidget(selection_button_configuration)
+        subframe_configuration.setLayout(sublayout_configuration)
+        self.plot_interface_layout.addWidget(subframe_configuration)
+
+    def _hand_button_released(self):
+        sending_button = self.sender()
+        self._user_selections["Hand"] = str(sending_button.objectName())
+
+    def _finger_button_released(self):
+        sending_button = self.sender()
+        self._user_selections["Finger"] = str(sending_button.objectName())
+
+    def _joint_button_released(self):
+        sending_button = self.sender()
+        self._user_selections["Joint"] = str(sending_button.objectName())
+
+    def _configuration_button_released(self):
+        sending_button = self.sender()
+        self._user_selections["Configuration"] = str(sending_button.objectName())
+
+
+class CreatePlotConfigurations():
+    """
+    Dinamically create plot configuration
+    """
+    def __init__(self, rows, columns, configuration_name):
+        self._plots = []
+        self._plot_rows = []
+        self._plot_columns = []
+        self._xml_config_dir = os.path.expanduser("~/projects/shadow_robot/base_deps/src/sr_visualization_common/"
```

Comment:
please use ROS API to determine package location and build relative path from there